### PR TITLE
CODENVY-742: Enable support for multiple AWS ECR registries per install

### PR DESCRIPTION
**Pull request by mmorhun**
### What does this PR do?
This PR adds ability to configure multiple AWS ECRs in properties files.

### What issues does this PR fix or reference?
#742 

### Previous Behavior
Only one AWS ECR can be configured.

### New Behavior
Multiple AWS ECRs can be configured.

### Tests written?
Yes

### CODE REVIEW

**Comment by mmorhun** [comment]
@garagatyi @skabashnyuk take a look, please

File: plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/auth/aws/ecr/AwsEcrAuthResolver.java [hunk 1]
```diff
@@ -65,29 +66,32 @@ public AwsEcrAuthResolver(AwsInitialAuthConfig awsInitialAuthConfig) {
...
+                    String authorizationToken = getAwsAuthorizationToken(awsAccountCredentials.getAccessKeyId(),
+                                                                         awsAccountCredentials.getSecretAccessKey());
+                    if (authorizationToken != null) {
+                        String decodedAuthorizationToken = new String(Base64.getDecoder().decode(authorizationToken));
+                        int colonIndex = decodedAuthorizationToken.indexOf(':');
+                        if (colonIndex != -1) {
+                            return newDto(AuthConfig.class).withUsername(decodedAuthorizationToken.substring(0, colonIndex))
+                                                           .withPassword(decodedAuthorizationToken.substring(colonIndex + 1));
+                        } else {
+                            LOG.warn("Cannot retrieve ECR credentials from token");
```
Comment (garagatyi) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/auth/aws/ecr/AwsEcrAuthResolver.java]: Please log registry here

File: plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/auth/aws/ecr/AwsEcrAuthResolver.java [hunk 2]
```diff
@@ -65,29 +66,32 @@ public AwsEcrAuthResolver(AwsInitialAuthConfig awsInitialAuthConfig) {
...
+                        }
                     }
+                } catch (IllegalArgumentException e) {
+                    LOG.warn("Retrieved AWS ECR authorization token has invalid format");
```
Comment (garagatyi) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/auth/aws/ecr/AwsEcrAuthResolver.java]: Why not error? Please log registry here.

File: plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/auth/aws/ecr/AwsEcrAuthResolver.java [hunk 3]
```diff
@@ -96,19 +100,20 @@ public AuthConfig getXRegistryAuth(@Nullable String registry) {
...
+                dynamicAuthConfigs.put(registry, authConfig);
+            }
         }
 
         return dynamicAuthConfigs;
     }
 
     @VisibleForTesting
-    String getAwsAuthorizationToken() {
+    String getAwsAuthorizationToken(String keyId, String keyValue) {
```
Comment (garagatyi) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/auth/aws/ecr/AwsEcrAuthResolver.java]: I think that Amazon doesn't have such a term **registry key value**. Please use terminology used before: accessKey, accessKeyId or something else if needed.

**Comment by mmorhun** [comment]
@skabashnyuk I had fixed it

**Pull request merged**